src/model/RacingGame.js

```diff
@@ -0,0 +1,62 @@
+const RacingCar = require('./RacingCar');
+
+class RacingGame {
+  #cars;
+  #winners;
+
+  constructor(names) {
+    this.#cars = [];
+    for (const name of names) {
+      this.#cars.push(new RacingCar(name));
+    }
+    this.#winners = [];
+  }
+
+  move() {
+    for (const car of this.#cars) {
+      car.move();
+    }
+  }
+
+  getCarNames() {
+    const carNames = this.#cars.map((car) => {
+      return car.getName();
+    });
+    return carNames;
+  }
+
+  getCarMoveResults() {
+    const carMoveResults = this.#cars.map((car) => {
+      return car.getMoveResult();
+    });
+    return carMoveResults;
+  }
+
+  calculateWinners() {
+    const carNames = this.getCarNames();
+    const carMoveResults = this.getCarMoveResults();
+    const carMoveResultLengths = getArrayLengths(carMoveResults);
+    const winningScore = Math.max(...carMoveResultLengths);
+    this.#setWinners(carNames, carMoveResultLengths, winningScore);
+  }
```

Comment:
```suggestion
  calculateWinners() {
    const carNames = this.getCarNames();
    const carMoveResults = this.getCarMoveResults();
    const winningScore =  Math.max(...carMoveResults.map(result => result.length));
        
    this.#setWinners(carNames, carMoveResultLengths, winningScore);
  }
```
최대한 로직을 나타내는 키워드는 지양해야 돼.
carMoveResultLengths라는 거는 결과의 길이를 가져와서 그걸로 winningScore를 구하겠다는 거잖아?
근데 사실 그거는 불필요한 정보야. winningScore를 길이로 구하건 굵기로 구하건 읽는 사람은 그냥 winningScore가 누군지가 궁금할 뿐이야.
그게 변수의 역할이야. 변수의 흐름만 봐도 로직이 어떻게 흘러가는지 알 수 있어야 돼. 승자 구하는 방식이 궁금하면 변수에 할당된 함수를 읽어보면 되는거고